Ten 115 - added enrolledAfterReportingFrraud page

Chrissystk opened the pull request:
Screenshot of Page for user coming from Fraud reporting
<img width="1072" alt="Screenshot 2022-04-08 at 15 16 37" src="https://user-images.githubusercontent.com/18244373/162454736-0265100a-1f68-4227-bbf5-8a837a76c333.png">

Screenshot of page for user choosing separate pt and sa
<img width="1072" alt="Screenshot 2022-04-08 at 15 15 56" src="https://user-images.githubusercontent.com/18244373/162454822-778a5399-2e17-4bcf-9d61-a126eb288759.png">

To view pages (after running tests to ensure sa user present in databases)
1. go to auth wizard and put credId = 9308587656991700, redirectUrl = http://localhost:7750/tax-enrolment-assignment-frontend/no-pt-enrolment?redirectUrl=http://localhost:7750/tax-enrolment-assignment-frontend/test-only/successful, nino = JS271436A and groupId =6949B2D7-BC9B-4E12-8411-3089E6148F10
2. Press submit and you should be presented with SA on other account.
3. Then manually put the url http://localhost:7750/tax-enrolment-assignment-frontend/enrol-pt/enrolment-success-sa-access-not-wanted to see when user chooses to have SA and PT separate.
4. Manually put the url http://localhost:7750/tax-enrolment-assignment-frontend/enrol-pt/report-suspicious-id
5. click continue and the page can be seen for after reporting fraud.

Ufuoma-Daniel commented on app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/EnrolledPTWithSAOnOtherAccountController.scala:
```diff
@@ -0,0 +1,73 @@
+    } yield (currentAccount.userId, optSAAccount.map(_.userId))
+
+    res.value.map {
+      case Right((currentUserId, optSAUserId)) =>
+        Ok(enrolledForPTPage(currentUserId, optSAUserId))
+      case Left(InvalidUserType(redirectUrl)) if redirectUrl.isDefined =>
+        Redirect(routes.AccountCheckController.accountCheck(redirectUrl.get))
+      case Left(_) => InternalServerError
```
Can we introduce some logs for the internal server error scenarios and these redirects to AccountCheckController.

Ufuoma-Daniel commented on app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/ReportSuspiciousIdController.scala:
```diff
@@ -21,46 +21,70 @@ import play.api.Logger
+      .checkValidAccountTypeAndEnrolForPT(SA_ASSIGNED_TO_OTHER_USER)
+      .value
+      .map {
+        case Right(_) =>
+          Redirect(routes.EnrolledPTWithSAOnOtherAccountController.view)
+        case Left(InvalidUserType(redirectUrl)) if redirectUrl.isDefined =>
+          Redirect(routes.AccountCheckController.accountCheck(redirectUrl.get))
+        case Left(UnexpectedResponseFromTaxEnrolments) =>
+          Ok(
+            errorView(
+              "enrolmentError.title",
+              "enrolmentError.heading",
+              "enrolmentError.body"
+            )
+          )
```
I won't mention it for every instance but the same logs should be added here.

Ufuoma-Daniel commented on app/uk/gov/hmrc/taxenrolmentassignmentfrontend/orchestrators/MultipleAccountsOrchestrator.scala:
```diff
@@ -25,24 +25,37 @@ import uk.gov.hmrc.service.TEAFResult
+  silentAssignmentService: SilentAssignmentService
 ) {
 
   def getDetailsForLandingPage(
```
Just for consistency can you change this from LandingPage to NotificationPage, ty!

Ufuoma-Daniel commented on conf/app.routes:
```diff
@@ -2,10 +2,13 @@
 GET        /no-pt-enrolment   uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.AccountCheckController.accountCheck(redirectUrl: String)
-GET        /enrol-pt/introduction   uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.LandingPageController.view
-POST       /enrol-pt/introduction   uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.LandingPageController.continue
+GET        /enrol-pt/introduction   uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.EnrolledForPTController.view
+POST       /enrol-pt/introduction   uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.EnrolledForPTController.continue
```
This route name isn't in the document anymore, needs updating.

Ufuoma-Daniel commented on app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/EnrolledPTWithSAOnOtherAccountController.scala:
```diff
@@ -0,0 +1,73 @@
+        Ok(enrolledForPTPage(currentUserId, optSAUserId))
+      case Left(InvalidUserType(redirectUrl)) if redirectUrl.isDefined =>
+        Redirect(routes.AccountCheckController.accountCheck(redirectUrl.get))
+      case Left(_) => InternalServerError
```
These INTERNAL_SERVER_ERRORS that are being thrown, what happens for the user? What do they see? The tests only look to be checking the status. I'm assuming it's a default error page but can we confirm that and have it reflected in the IT tests?

Chrissystk replied:
I think it will just be a blank page. We are awaiting the story for error pages.

Ufuoma-Daniel replied:
We already have the error page template, it might be better to redirect to that with the default error text.

Chrissystk commented on test/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/messages/ReportSuspiciousIDMessages.scala:
```diff
@@ -19,15 +19,20 @@ package uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.messages
     "Please download a copy of this information to your device " +
     "or make a note of it as this screen may time out after 15 minutes."
-  val linkParagraphText: String = "download a copy of this information to your device"
+  val linkParagraphText: String =
+    "download a copy of this information to your device"
```
It's the scalafmt formatter. It auto formats that way.

Ufuoma-Daniel commented:
Last thing, we have moved from the LandingPage naming to be consistent with everywhere else, but the new name NotificationPage isn't the same as EnrolledForPTPage. We might just be persisting the confusion. Can we rename to EnrolledNotificationPage or some sort of middle ground?

Ufuoma-Daniel commented on conf/messages:
```diff
@@ -8,14 +8,14 @@ enrolCurrentUser.currentUserId = My current user ID {0}
-landingPage.continue = Continue
+enrolledForPT.title = Only your current user ID can access your personal tax account from now on
+enrolledForPT.heading = Only your current user ID can access your personal tax account from now on
+enrolledForPT.paragraph1 = You are currently signed in with the Government Gateway user ID {0}.
```
This should be "You are currently signed in with the Government Gateway user ID **ending with** {0}.

Chrissystk replied:
This is a change from the AccountDetails model. We are having a separate story to cover it.

Chrissystk replied:
The model itself will say ending with

Chrissystk replied:
Either that or we have to update it in all the views

Ufuoma-Daniel commented on conf/messages:
```diff
@@ -64,6 +64,13 @@ signInAgain.title = Sign in again
 signInAgain.paragraph = with the Government Gateway user ID you want to use to access your personal tax information.
+enrolledForPTWithSAOnOtherAccount.title = Only your current user ID can access your personal tax account from now on
+enrolledForPTWithSAOnOtherAccount.heading = Only your current user ID can access your personal tax account from now on
+enrolledForPTWithSAOnOtherAccount.paragraph1 = You are currently signed in with the Government Gateway user ID {0}.
```
Same here, "ending with"

Chrissystk commented:
https://github.com/hmrc/tax-enrolment-assignment-journey-tests/pull/25 to reflect new urls

I have added logs, updated urls and used common functions for PT on other account.

Ufuoma-Daniel commented on app/uk/gov/hmrc/taxenrolmentassignmentfrontend/views/ReportSuspiciousID.scala.html:
```diff
@@ -86,14 +88,15 @@ <h1 class="govuk-heading-xl">@{messages(s"ReportSuspiciousID.heading")}</h1>
-
+@if(saOnOtherAccount) {
+  @formHelper(action = routes.ReportSuspiciousIDController.continue) {
+    <p class="govuk-body">@messages("ReportSuspiciousID.sa.text1") </p>
```
Isn't this wrong? Shouldn't it have stayed as '`currentAccountHasSA'`? The message key is for the message - 

> You can continue to your personal tax account with the user ID you are currently signed in with.

Chrissystk replied:
No this is reporting fraud after going through the SA on another account journey

Ufuoma-Daniel replied:
There are two ways to get to the report fraud page. Either for an unknown account having the PT or SA enrolment. So If the user has come through the PT report journey, but has SA on another account, this content would appear and be invalid. If anything, this text shouldn't be dependent on SA but rather the PT enrolment.

Chrissystk replied:
Yes an unknown account not the current account

Chrissystk replied:
It is the saOnOtherAccount journey that this button is shown for.

Ufuoma-Daniel replied:
I don't know if I've explained myself well, what I'm saying is that this text should only appear if your `currentAccountHasPT`, because otherwise the text that appears will be wrong sometimes.

Chrissystk replied:
I've changed it to saOnOtherAccountJourney

Chrissystk replied:
No if an account has already been assigned PT they can not then continue and assign a PT enrolment to their current account. The only way they can continue is if a enrolment hasn't already been assigned, i.e the case where they don't recognise the credential that has a SA enrolment.

Chrissystk replied:
<img width="1526" alt="Screenshot 2022-04-13 at 10 59 00" src="https://user-images.githubusercontent.com/18244373/163153925-9b494ee9-654b-4bfa-925a-d74f029857f6.png">
